src/Symfony/Component/Console/Parallelization/Parallelization.php

```diff
@@ -0,0 +1,475 @@
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Process\PhpExecutableFinder;
+
+/**
+ * Adds parallelization capabilities to console commands.
+ *
+ * Make sure to call configureParallelization() in your configure() method!
+ *
+ * You must implement the following methods in your command:
+ *
+ *  * fetchElements(): Returns all the elements that you want to process as
+ *    strings. Typically, you will return IDs of database objects here.
+ *  * runSingleCommand(): Executes the command for a single element.
+ *  * getElementName(): Returns a human readable name of the processed things
+ *
+ * You can improve the performance of your command by making use of batching.
+ * Batching allows you to process multiple elements together, for example to
+ * persist them in a batch to reduce the number of I/O operations.
+ *
+ * To enable batching, you will typically implement runAfterBatch() and persist
+ * the changes done in multiple calls of runSingleCommand().
+ *
+ * The batch size is determined by getBatchSize() and defaults to the segment
+ * size. The segment size is the number of elements a worker (child) process
+ * consumes before it dies. This means that, by default, a child process will
+ * process all its elements, persist them in a batch and then die. If you want
+ * to improve the performance of your command, try to tweak getSegmentSize()
+ * first. Optionally, you can tweak getBatchSize() to process multiple batches
+ * in each child process.
+ */
+trait Parallelization
+{
+    /**
+     * Returns the symbol for communicating progress from the child to the
+     * main process.
+     *
+     * @return string A single character
+     */
+    private static function getProgressSymbol()
+    {
+        return chr(254);
+    }
+
+    /**
+     * Detects the path of the PHP interpreter.
+     *
+     * @throws RuntimeException If PHP could not be found
+     *
+     * @return string The absolute path to the PHP interpreter
+     */
+    private static function detectPhpExecutable(): string
+    {
+        $executableFinder = new PhpExecutableFinder();
+
+        $php = $executableFinder->find();
+
+        if (false === $php) {
+            throw new RuntimeException('Cannot find php executable');
+        }
+
+        return $php;
+    }
+
+    /**
+     * Returns the environment variables that are passed to the child processes.
+     *
+     * @param ContainerInterface $container The service containers
+     *
+     * @return string[] A list of environment variable names and values
+     */
+    private static function getEnvironmentVariables(ContainerInterface $container): array
+    {
+        return [
+            'PATH' => getenv('PATH'),
+            'HOME' => getenv('HOME'),
+            'SYMFONY_DEBUG' => $container->getParameter('kernel.debug'),
+            'SYMFONY_ENV' => $container->getParameter('kernel.environment'),
```

Comment:
Not AFAIR. I wouldn't have added this if it hadn't been necessary.

Reply:
Well, the environment is inherited by default since a 3.x release (I think it was 3.2 but not sure). Older releases were requiring to opt-in. On which Symfony version was this code written initially ?